Scripts/Runtime/Job/AccessControl/AccessController.cs

```diff
@@ -157,6 +186,7 @@ public JobHandle AcquireAsync(AccessType accessType)
                 default:
                     throw new ArgumentOutOfRangeException(nameof(accessType), accessType, $"Tried to acquire with {nameof(AccessType)} of {accessType} but no code path satisfies!");
             }
+            Debug.Assert(acquiredHandle.Equals(GetDependencyFor(accessType)));
```

Comment:
Minor but I feel like this is better used as a Conditional method check under `ENABLE_UNITY_COLLECTIONS_CHECKS`. Seems too heavy for a Debug Assert.

Reply:
Going to leave it for now. Discussed a bit and
 - Really this should get handled by a Unit test and removed. #129 
 - This is a cheap check
 - Collections checks tend to be correct usage type checks/assertions while this check is more of a "is our logic consistent" check.

I did add a TODO reference comment for the assert.